src/pg_query_summary.c

```diff
@@ -0,0 +1,761 @@
+		}
+
+		case T_AlterTableStmt: {
+			AlterTableStmt* stmt = castNode(AlterTableStmt, node);
+
+			// `ALTER INDEX index_name` is ignored.
+			if (stmt->objtype != OBJECT_INDEX)
+				add_from_clause_item(stmt->relation, CONTEXT_DDL, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateStmt:
+			add_from_clause_item(castNode(CreateStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_CreateTableAsStmt: {
+			CreateTableAsStmt* stmt = castNode(CreateTableAsStmt, node);
+			if (stmt->into && stmt->into->rel)
+				add_from_clause_item(stmt->into->rel, CONTEXT_DDL, state);
+
+			if (stmt->query)
+				pg_query_summary_walk_impl(stmt->query, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_TruncateStmt:
+			add_from_clause_items(castNode(TruncateStmt, node)->relations, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_ViewStmt: {
+			ViewStmt* stmt = castNode(ViewStmt, node);
+			add_from_clause_item(stmt->view, CONTEXT_DDL, state);
+			pg_query_summary_walk_impl(stmt->query, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_IndexStmt: {
+			IndexStmt* stmt = castNode(IndexStmt, node);
+
+			add_from_clause_item(stmt->relation, CONTEXT_DDL, state);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->indexParams) {
+				IndexElem* index_elem = lfirst(lc);
+				if (index_elem->expr)
+					pg_query_summary_walk_impl(index_elem->expr, state);
+			}
+
+			if (stmt->whereClause)
+				pg_query_summary_walk_impl(stmt->whereClause, state);
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateTrigStmt:
+			add_from_clause_item(castNode(CreateTrigStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_RuleStmt: {
+			add_from_clause_item(castNode(RuleStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_VacuumStmt: {
+			VacuumStmt* stmt = castNode(VacuumStmt, node);
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->rels) {
+				VacuumRelation* rel = lfirst(lc);
+				add_from_clause_item(rel->relation, CONTEXT_DDL, state);
+			}
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_RefreshMatViewStmt:
+			add_from_clause_item(castNode(RefreshMatViewStmt, node)->relation, CONTEXT_DDL, state);
+			return false; // Not handled by tree walker.
+
+		case T_FuncCall:
+			add_function(castNode(FuncCall, node)->funcname, CONTEXT_CALL, state);
+			break;
+
+		case T_DropStmt: {
+			DropStmt* stmt = castNode(DropStmt, node);
+
+			switch (stmt->removeType) {
+				case OBJECT_TABLE: {
+					ListCell* lc = NULL;
+					foreach(lc, stmt->objects) {
+						Node* obj = lfirst(lc);
+						if (obj->type == T_List) {
+							List* list = castNode(List, obj);
+
+							size_t len = list_length(list);
+							if (len == 0)
+								continue;
+
+							if (len == 1)
```

Comment:
We may also have do deal with `len = 3` here, in case of a table qualified by both schema and database name (`DROP TABLE db.schema.x`)

Reply:
(this would resolve itself if we switch to makeRangeVarFromNameList)